src/ClientData/TimerDataInterface.cpp

```diff
@@ -15,16 +15,9 @@ uint64_t GetNextPixelBoundaryTimeNs(uint64_t current_timestamp_ns, uint32_t reso
   uint64_t current_pixel = (current_ns_from_start * resolution) / total_ns;
   uint64_t next_pixel = current_pixel + 1;
 
-  // To calculate the timestamp of a pixel boundary, we round to the left similar to how it works in
-  // other parts of Orbit.
-  uint64_t next_pixel_ns_from_min = total_ns * next_pixel / resolution;
-
-  // Border case when we have a lot of pixels who have the same timestamp (because the number of
-  // pixels is less than the nanoseconds in screen). In this case, as we've already drawn in the
-  // current_timestamp, the next pixel to draw should have the next timestamp.
-  if (next_pixel_ns_from_min == current_ns_from_start) {
-    next_pixel_ns_from_min = current_ns_from_start + 1;
-  }
+  // To calculate the timestamp of a pixel boundary, we make a ceiling to be consistent to how we
+  // calculate current_pixel.
+  uint64_t next_pixel_ns_from_min = (total_ns * next_pixel + resolution - 1) / resolution;
```

Comment:
I might be a bit slow today, but I really dont understand this. Maybe you can extract parts of this in a more readable way like this:
```
// Calculates `ceil(dividend / divisor)` only using integers 
const auto rounding_up_division = [](uint64_t dividend, uint64_t divisor) {
  return (dividend + divisor - 1) / divisor;
}
uint64_t next_pixel_ns_from_min = rounding_up_division(total_ns * next_pixel, resolution);
```

You could also link http://www.cs.nott.ac.uk/~psarb2/G51MPC/slides/NumberLogic.pdf which explains this

Reply:
Added your suggestion. Modified to a `1 + (dividend - 1) / divisor;` that has an issue with dividend 0 but not with overflow (it was failing in ThreadTrackDataProviderTest).